### PR TITLE
Update Image.php

### PR DESCRIPTION
**Pull request by gulaandrij**
fix broken image if service unavailable

### CODE REVIEW

**Comment by gulaandrij** [comment]
any feedback on this?

**Comment by fzaninotto** [comment]
Thanks!

File: src/Faker/Provider/Image.php [hunk 1]
```diff
@@ -74,9 +74,15 @@ public static function image($dir = null, $width = 640, $height = 480, $category
             $fp = fopen($filepath, 'w');
             $ch = curl_init($url);
             curl_setopt($ch, CURLOPT_FILE, $fp);
-            $success = curl_exec($ch);
+            $success = curl_exec($ch) || curl_getinfo($ch, CURLINFO_HTTP_CODE) === 200;
+
+            if ($success){
+                fclose($fp);
```
Comment (ethanclevenger91) [inline src/Faker/Provider/Image.php]: Shouldn't this happen regardless of success?

File: src/Faker/Provider/Image.php [hunk 2]
```diff
@@ -74,9 +74,15 @@ public static function image($dir = null, $width = 640, $height = 480, $category
...
+            } else {
+                unlink($filepath);
+            }
+
             curl_close($ch);
-            fclose($fp);
         } elseif (ini_get('allow_url_fopen')) {
```
Comment (ethanclevenger91) [inline src/Faker/Provider/Image.php]: Shouldn't the same checks go here in case `curl` isn't enabled?

**Comment by gulaandrij** [comment]
> Silently deleting the image seems like unexpected behavior.
if lorempixel is not available then the image is saved as spoiled, 0 kb, so there is no need to store it

**Comment by ethanclevenger91** [comment]
Yeah, I get that much. But if I expect all my images to be generated, some are inexplicably missing, and I'm not versed in the intricacies of this issue, what do I do?

**Comment by gulaandrij** [comment]
in my opinion its a good catch to add additional check 
```
if (!$faker->image()) {

} 
```

**Comment by ethanclevenger91** [comment]
But once again, if I'm a user who has no idea this is an issue, how would I know to do that?

I certainly don't do anything like:
```
if (!$faker->text()) {

}
```

**Pull request merged**